packages/@aws-cdk/aws-rds/lib/instance.ts

```diff
@@ -1022,33 +997,25 @@ export class DatabaseInstanceFromSnapshot extends DatabaseInstanceSource impleme
   constructor(scope: Construct, id: string, props: DatabaseInstanceFromSnapshotProps) {
     super(scope, id, props);
 
-    let secret: DatabaseSecret | undefined;
-
-    if (props.generateMasterUserPassword) {
-      if (!props.masterUsername) { // We need the master username to include it in the generated secret
-        throw new Error('`masterUsername` must be specified when `generateMasterUserPassword` is set to true.');
-      }
-
-      if (props.masterUserPassword) {
-        throw new Error('Cannot specify `masterUserPassword` when `generateMasterUserPassword` is set to true.');
+    let login = props.masterUser;
+    let secret = login?.secret;
+    if (!secret && login?.generatePassword) {
+      if (!login.username) {
+        throw new Error('`username` must be specified when `generatePassword` is set to true');
```

Comment:
I feel like this error is rather cryptic. Will a user of `DatabaseInstanceFromSnapshot` understand what's gone wrong here, and know how to unblock themselves?

Reply:
What's cryptic about it? Either they used the factory methods, in which case this error is impossible, or they've manually created a Login object with `generatePassword` true and `username` blank. This error tells them to set the username. 

What's your suggestion for an alternative?